### PR TITLE
Support more connectors

### PR DESCRIPTION
**Pull request by LucGenetier**
- ignore x-ms-notification-content
- ignore x-ms-capabilities
- ignore x-ms-test-value
- ignore x-ms-url-encoding
- improve TestAllConnectors test
- do not throw an exception when the return type isn't supported but use IsSupported flag instead

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Robulane — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -357,3 +357,11 @@ tool/
...
 # ApiCompat suppression files
 suppress.*.xml
+
+# Connetor analysis reports
+Analysis.txt
+Report.json
+Report_*.json
+
+# global.json
+global.json
```
Comment (MikeStall) [inline .gitignore:367]: what file is this?

File: .gitignore [hunk 2]
```diff
@@ -357,3 +357,11 @@ tool/
 
 # ApiCompat suppression files
 suppress.*.xml
+
+# Connetor analysis reports
+Analysis.txt
+Report.json
+Report_*.json
```
Comment (MikeStall) [inline .gitignore:364]: could we ignore by directory?

File: .gitignore (hunk 1 above)
Comment (LucGenetier) [reply .gitignore:367]: [https://learn.microsoft.com/en-us/dotnet/core/tools/global-json](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json)

It's set during build pipeline but I don't want it to be checked-in if I define it locally

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**